framework/include/vectorpostprocessors/LineMaterialSamplerBase.h

```diff
@@ -179,6 +184,7 @@ LineMaterialSamplerBase<T>::execute()
       addSample(_q_point[qp], qp_proj_dist_along_line, values);
     }
   }
+  _fe_problem.clearActiveMaterialProperties(_tid);
```

Comment:
There is a lot of this pattern (e.g in all the compute threads classes) - setting up active material properties state and then clearing it with important code in between.  I'd be worried about this being difficult to keep track of - easy for bugs to appear.  Someone might skip the clear with an early return for example.  Or someone is surprised when there material properties aren't being recomputed as expected because they didn't know they needed to go through this explicit property id adding process.

* Could we somehow make this safer so that people don't get surprised by that?
* Is there a better place we could put the clear to be more "automatic" - and possible a place that would remove the need for certain classes to have to re-add active properties temporarily like here?